docs/program/apis/data-client.md

```diff
@@ -0,0 +1,246 @@
+- `data` [(Optional[bytes])](https://docs.python.org/3/library/stdtypes.html#bytes-objects): Optional bytes representing file data to upload.
+
+**Returns**:
+
+- [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): ID of the new file.
+
+```python {class="line-numbers linkable-line-numbers"}
+file_id = await data_client.file_upload(
+    part_id="INSERT YOUR PART ID",
+    component_type='motor',
+    component_name='left_motor',
+    method_name='IsPowered',
+    tags=["tag_1", "tag_2"],
+    file_name="your-file",
+    file_extension=".txt"
+)
+```
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/app/data_client/index.html#viam.app.data_client.DataClient.file_upload).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### FileUploadFromPath
+
+Upload files stored on your machine to the [Viam app](https://app.viam.com) by filepath.
+Uploaded files can be found under the **Files** subtab of the app's [**Data** tab](https://app.viam.com/data).
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- `filepath` [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): The absolute filepath of the file to be uploaded.
+- `part_id` [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Part ID of the component used to capture the data. See [Find Part ID](#find-part-id) for instructions on retrieving this value.
+- `component_type` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Type of the component used to capture the data.
+- `component_name` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Name of the component used to capture the data.
+- `method_name` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Name of the method used to capture the data.
+- `tags` [(Optional[List[str]])](https://docs.python.org/3/library/stdtypes.html#typesseq-list): Optional list of [image tags](/manage/data/label/#image-tags) to allow for tag-based data filtering when retrieving data.
+
+**Returns**:
+
+- [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): ID of the new file.
+
+```python {class="line-numbers linkable-line-numbers"}
+file_id = await data_client.file_upload_from_path(
+    part_id="INSERT YOUR PART ID",
+    component_type='motor',
+    component_name='left_motor',
+    method_name='IsPowered',
+    tags=["tag_1", "tag_2"],
+    filepath="/Users/<your-username>/<your-directory>/<your-file.txt>"
+)
+```
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/app/data_client/index.html#viam.app.data_client.DataClient.file_upload_from_path).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Find Part ID
```

Comment:
lol this is terrible - glad this will also be fixed soon

Reply:
Yep 😆

Reply:
Ahhhhhhhhhhhhhh

Reply:
(but this is the best possible for what we are given -- good find!)

Reply:
Up to you, but might be worth mentioning the quicker way to retrieve this if you have images/files uploaded!

Reply:
But that won't be relevant if it's your first time uploading data, so maybe ok to omit.

Reply:
Yeah, I couldn't see anything when I hadn't uploaded yet-- no way of knowing if that'll be the case for most users but I figure we might as well instruct them consistently, even if it is a bit hacky.